enginetest/priv_auth_queries.go

```diff
@@ -1662,4 +1662,242 @@ var QuickPrivTests = []QuickPrivilegeTest{
 			"UPDATE mydb.test SET v1 = 0;",
 		},
 	},
+	{
+		Queries: []string{
+			"FLUSH PRIVILEGES;",
+		},
+		ExpectingErr: true,
+	},
+	{
+		Queries: []string{
+			"GRANT RELOAD ON *.* TO tester@localhost",
+			"FLUSH PRIVILEGES;",
+		},
+	},
+}
+
+// PersistPrivilegesTest test the persist function in the grant tables, and we'll run an INSERT, UPDATE,
+// or DELETE on one of the privilege tables (mysql.user, mysql.db, etc.), modifying the privileges in some way.
+type PersistPrivilegesTest struct {
+	Name        string
+	SetUpScript []string
+	InOneServer []PersistPrivilegesTestInOneServer
+}
+
+// PersistPrivilegesTestInOneServer test persist assertions with restarting the server with the same privileges file
+type PersistPrivilegesTestInOneServer struct {
+	Assertions []PersistPrivilegesTestAssertion
+}
+
+// PersistPrivilegesTestAssertion is within a PersistPrivilegesTestInOneServer to assert functionality.
+type PersistPrivilegesTestAssertion struct {
+	User           string
+	Host           string
+	Query          string
+	Expected       []sql.Row
+	ExpectedErr    *errors.Kind
+	ExpectedErrStr string
+}
+
+// PersistPrivilegesTests test persistence of privilege systems. These tests always have the root account available, and the root
+// account is used with any queries in the SetUpScript.
+var PersistPrivilegesTests = []PersistPrivilegesTest{
```

Comment:
Reading these tests, it's not clear exactly how `FLUSH PRIVILEGES` work. You explained it to me so I understand the intent, but it's a bit more complicated than the easier alternative. It is not required that we follow this script style for tests. This style is mainly for writing a lot of different tests in a clear and easy way. In this case we're only testing a single thing, which is whether `FLUSH PRIVILEGES` actually flushes privileges, so everything can be inside of the test function.